phpBB/viewonline.php

```diff
@@ -32,491 +33,30 @@
-		ORDER BY g.' . $order_legend . ' ASC';
-}
-$result = $db->sql_query($sql);
 
-$legend = '';
-while ($row = $db->sql_fetchrow($result))
-{
-	if ($row['group_name'] == 'BOTS')
-	{
-		$legend .= (($legend != '') ? ', ' : '') . '<span style="color:#' . $row['group_colour'] . '">' . $user->lang['G_BOTS'] . '</span>';
-	}
-	else
-	{
-		$legend .= (($legend != '') ? ', ' : '') . '<a style="color:#' . $row['group_colour'] . '" href="' . append_sid("{$phpbb_root_path}memberlist.$phpEx", 'mode=group&amp;g=' . $row['group_id']) . '">' . $group_helper->get_name($row['group_name']) . '</a>';
-	}
-}
-$db->sql_freeresult($result);
-
-// Refreshing the page every 60 seconds...
-meta_refresh(60, append_sid("{$phpbb_root_path}viewonline.$phpEx", "sg=$show_guests&amp;sk=$sort_key&amp;sd=$sort_dir&amp;start=$start"));
-
-$start = $pagination->validate_start($start, $config['topics_per_page'], $counter);
-$base_url = append_sid("{$phpbb_root_path}viewonline.$phpEx", "sg=$show_guests&amp;sk=$sort_key&amp;sd=$sort_dir");
-$pagination->generate_template_pagination($base_url, 'pagination', 'start', $counter, $config['topics_per_page'], $start);
-
-$template->assign_block_vars('navlinks', array(
-	'BREADCRUMB_NAME'	=> $user->lang('WHO_IS_ONLINE'),
-	'U_BREADCRUMB'		=> append_sid("{$phpbb_root_path}viewonline.$phpEx"),
-));
-
-// Send data to template
-$template->assign_vars(array(
-	'TOTAL_REGISTERED_USERS_ONLINE'	=> $user->lang('REG_USERS_ONLINE', (int) $logged_visible_online, $user->lang('HIDDEN_USERS_ONLINE', (int) $logged_hidden_online)),
-	'TOTAL_GUEST_USERS_ONLINE'		=> $user->lang('GUEST_USERS_ONLINE', (int) $guest_counter),
-	'LEGEND'						=> $legend,
-
-	'U_SORT_USERNAME'		=> append_sid("{$phpbb_root_path}viewonline.$phpEx", 'sk=a&amp;sd=' . (($sort_key == 'a' && $sort_dir == 'a') ? 'd' : 'a') . '&amp;sg=' . ((int) $show_guests)),
-	'U_SORT_UPDATED'		=> append_sid("{$phpbb_root_path}viewonline.$phpEx", 'sk=b&amp;sd=' . (($sort_key == 'b' && $sort_dir == 'a') ? 'd' : 'a') . '&amp;sg=' . ((int) $show_guests)),
-	'U_SORT_LOCATION'		=> append_sid("{$phpbb_root_path}viewonline.$phpEx", 'sk=c&amp;sd=' . (($sort_key == 'c' && $sort_dir == 'a') ? 'd' : 'a') . '&amp;sg=' . ((int) $show_guests)),
-
-	'U_SWITCH_GUEST_DISPLAY'	=> append_sid("{$phpbb_root_path}viewonline.$phpEx", 'sg=' . ((int) !$show_guests)),
-	'L_SWITCH_GUEST_DISPLAY'	=> ($show_guests) ? $user->lang['HIDE_GUESTS'] : $user->lang['DISPLAY_GUESTS'],
-	'S_SWITCH_GUEST_DISPLAY'	=> ($config['load_online_guests']) ? true : false,
-	'S_VIEWONLINE'				=> true,
-));
-
-// We do not need to load the who is online box here. ;)
-$config['load_online'] = false;
-
-// Output the page
-page_header($user->lang['WHO_IS_ONLINE']);
-
-$template->set_filenames(array(
-	'body' => 'viewonline_body.html')
+$response = new RedirectResponse(
+	$controller_helper->route('phpbb_members_online', array(
+		'mode'	=> $mode,
+		's'		=> $session_id,
+		'start'	=> $start,
```

Comment:
You removed all those variables, they won't be passed to the redirected URL. Ideally you should just `$request->variable('start', 0)` all of them.

Reply:
I don't understand, they haven't been removed

Reply:
Oh sorry, I just didn't see them. :)